src/packages.ts

```diff
@@ -436,39 +434,41 @@ export class RockCXXPackage extends RockPackage
 
 export class RockOrogenPackage extends RockPackage
 {
-    private _bridge: async.EnvironmentBridge;
-
-    constructor(bridge: async.EnvironmentBridge, ws: autoproj.Workspace, info: autoproj.Package, context: context.Context, vscode: wrappers.VSCode)
-    {
-        super(ws, info, context, vscode);
-        this._bridge = bridge;
-    }
-
-    async pickTask(): Promise<async.IOrogenTask | undefined>
+    async pickTask(): Promise<syskit.AvailableDeployment | undefined>
     {
-        let description = this._bridge.describeOrogenProject(this.path, basename(this.path));
+        let syskitConnection = await this.workspace.
+            syskitDefaultConnection(this._vscode);
+        let deployments = syskitConnection.availableDeployments();
         let tokenSource = new vscode.CancellationTokenSource();
 
         let err;
-        description.catch((_err) => {
+        deployments.catch((_err) => {
             err = _err;
             tokenSource.cancel();
         })
-        let promise = description.then(
-            (result) => {
-                let choices: { label: string, description: string, task: async.IOrogenTask }[] = [];
-                result.forEach((task) => {
+        let promise = deployments.then((result) => {
+            let choices: any[] = [];
+
+            result.forEach((deployment) => {
```

Comment:
I think we should filter only tasks and deployments defined by the package. Ideally, syskit's rest api should provide this since it will probably be common among apps using the api.

Reply:
> I think we should filter only tasks and deployments defined by the package.

I don't. I'm going for stopping to decide what the developer might or might not want to do. This is vscode's philosophy, especially with the launch files, and so far it's a side of it I really love.

Reply:
Well, that's what editing launch.json is for (to let the developer do whatever he/she wants). The "Add launch entry" command is to keep things simple and convenient. But "i won't fight over this" :P